CODE_OF_CONDUCT.md

```diff
@@ -1,36 +1,114 @@
+
+Examples of other unacceptable behaviors by participants include but are not limited to:
 
 * The use of sexualized language or imagery and unwelcome sexual attention or
-advances
-* Trolling, insulting/derogatory comments, and personal or political attacks
-* Public or private harassment
+advances.
+* Violent threats or language directed against another person.
+* Posting sexually explicit or violent material.
+* Sexist, racist, homophobic, transphobic, ableist or otherwise discriminatory jokes and language.
+* Trolling, insulting/derogatory comments, and personal or political attacks particulary those related to gender, sexual orientation, race, religion or disability.
+* Public or private harassment.
 * Publishing others' private information, such as a physical or electronic
-  address, without explicit permission
-  * Other conduct which could reasonably be considered inappropriate in a
+  address, without explicit permission ("doxing")
+* Other conduct which could reasonably be considered inappropriate in a
     professional setting
 
+If you have any lack of clarity about behaviors we include in the definition of "harassment", please read the [Citizen Code of Conduct](http://citizencodeofconduct.org/). In particular, we don’t tolerate behavior that excludes people in socially marginalized groups.
+
+### Consequences of Unacceptable Behavior
+Unacceptable behavior from any community member, including sponsors and those with
+decision-making authority, will *not* be tolerated.
+
+Anyone asked to stop unacceptable behavior is expected to comply immediately.
+
+If a community member engages in unacceptable behavior, the community organizers
+may take action they deem appropriate, up to and including temporary ban or permanent
+expulsion from the community _without warning_ (and without refund in the case of
+a paid event).
+
+Any physical violence _or_ intimidation, threatened or acted on, is a serious
+offense and will result in immediate exclusion from the community and appropriate
+follow up with law enforcement (no, we're not kidding).
+
+### Handling Incidents
+When an advocate or maintainer notices someone behaving in a way that is outside of our guidelines (a violator) the advocate should make every reasonable attempt to help curtail that behavior.  The advocate may:
+
+* Remind the violator about our Community Code of Conduct and provide a link to this document.
+* Ask the violator to stop the unacceptable behavior.
+* Raise the issue with a maintainer, the community manager, or any member of the core project team.
+* Allow time for the violator to correct the behavior.
+
+The advocate should take the following steps if the behavior is not brought inline with our guidelines or the incident is not resolved.
+
+* Consult with another maintainer to make a judgement call about what reasonable corrective actions are warranted.
+* Apply the corrective action.
+* Document the incident.
+
```

Comment:
Should we put in a standard email or google voice number to call and call it out here? If we standardize on it, it could in the case of an incident to say "email this" "call that number" and you'll find someone to help.